packages/paste-core/utilities/grid/src/index.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import {ThemeShape} from '@twilio-paste/theme-tokens';
+import {ResponsiveValue} from 'styled-system';
+import {Box} from '@twilio-paste/box';
+import {Flex} from '@twilio-paste/flex';
+import {FlexboxProps, LayoutProps, MarginProps} from '@twilio-paste/style-props';
+
+export interface GridProps {
+  children: NonNullable<React.ReactNode>;
+  gutter?: ResponsiveValue<keyof ThemeShape['space']>;
+}
+
+const getColumns = ({children}: GridProps): number => {
+  return React.Children.count(children);
+};
+
+const Grid: React.FC<GridProps> = props => {
+  const Count = getColumns(props);
+
+  const GridColumns = React.Children.map(props.children, child =>
+    React.isValidElement(child) ? React.cloneElement(child, {count: Count, gutter: props.gutter}) : child
```

Comment:
Generally speaking being reliant on, inspecting, modifying components another component doesn't own isn't something we wanna do.

So, why shouldn't I be concerned about this?

Reply:
`React.cloneElement` is pretty safe from a performance standpoint. Its basically just copying an object and passing props. For a more in depth answer: https://stackoverflow.com/questions/54922160/react-cloneelement-in-list-performance